site/tutorials/streaming.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: tutorials
+menu: tutorials
+title: Streaming Data
+permalink: /tutorials/streaming.html
+---
+
+
+In this tutorial, you'll learn how to stream new data in a Vega-Lite visualization. If you are not familiar with Vega-Lite, please read the [getting started tutorial]({{site.baseurl}}/tutorials/getting_started.html) first.
+
+We will be using the `View` API from the Vega, where we update data via the [`change`](https://vega.github.io/vega/docs/api/view/#data) method.  Data in Vega and Vega-Lite is updated either via interactions, or by specifying the data changes directly.  In both cases, the data is pushed through the Vega runtime for optimized performance.  This tutorial focuses on the latter for streaming data, where no user interaction is involved.
+
+<div id="chart" class="vl-example"></div>
+
+To use `view.change`, you have to first specify the name of the data source you are udpating:  `view.change('data', ... )`.  The name needs to be specified in the original spec, as described [here]({{site.baseurl}}/docs/data.html#named).  As the second parameter to the `view.change` method, you should specify what data to add, and what data to remove.  To add data, simply supply the array of data entries matching previous schemas, for the example here, it is:
+```
+{
+  "x": number,
+  "y": number,
+  "category": number
+}
+```
+
+As for removing data points, you could either supply the actual data points to be removed similar to the insert above (which requires your storing the data), or you could filter existing rendered data by some attributes and pass the predicate to the `remove()` function, such as in this case `function (t) { return t.x < minimumX; }`.  If you are familiar with D3, this `insert` and `remove` pattern is similar to the [enter exit](https://bost.ocks.org/mike/circles/#entering) pattern.
+
+Putting it together, we have
+```
+  var changeSet = vega.changeset().insert(valueGenerator()).remove(function (t) { return t.x < minimumX; });
+  res.view.change('table', changeSet).run();
```

Comment:
Maybe better just write `view.change` and say that we will cover where the `view` comes from later.